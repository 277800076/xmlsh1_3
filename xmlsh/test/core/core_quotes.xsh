# Tests of basic quote parsing

# basic single and double quotes
echo foo bar
echo 'foo'
echo "foo bar"
echo 'foo bar'

echo foo"bar"
echo foo'bar'

echo "foo"bar
echo 'foo'bar

echo foo"bar"spam
set foo" bar "spam
echo $#
echo $*
set foo' bar 'spam
echo $#
echo $*


echo foo"bar'spam'bletch"barf


# variable expansion within quotes
_A=test

echo "$_A"
echo foo"$_A"
echo foo"$_A"bar
echo "$_A"foo


# verify expansion does NOT take place in single quotes
echo '$_A'
echo foo'$_A'
echo foo'$_A'bar

# Tests of backslashes

echo foo\bar
echo foo\\bar
echo foo\\\bar
echo 'foo\bar'
echo 'foo\\bar'
echo 'foo\\\bar'
echo "foo\bar"
echo "foo\\bar"
echo "foo\\\bar"

# Multiline quoting

echo "on a single \
line"

echo 'on a single \
line'

echo "line 1
line 2"

echo 'line 1
line 2'



# Test that we strip quotes off of variable assigments
_A="test"
echo <[ <foo attr="{$_A}"/> ]>
