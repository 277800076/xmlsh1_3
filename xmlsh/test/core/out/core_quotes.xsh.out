foo bar
foo
foo bar
foo bar
foobar
foobar
foobar
foobar
foobarspam
1
foo bar spam
1
foo bar spam
foobar'spam'bletchbarf
test
footest
footestbar
testfoo
$_A
foo$_A
foo$_Abar
foobar
foo\bar
foo\bar
foo\bar
foo\\bar
foo\\\bar
foo\bar
foo\bar
foo\\bar
on a single line
on a single line
line 1
line 2
line 1
line 2
<foo attr="test"/>
"" ""
2
"" ""
"" ""
"
'"
'foo bar'
